### PR TITLE
logging: stop using formatted strings for object identifiers

### PR DESCRIPTION
**Pull request by stevekuznetsov**
logging: format Kind better when none is present

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

logging: stop using formatted strings for object identifiers

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

Part of #1694 
/assign @sttts @ncdc

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Before:

```
I0810 11:22:39.327675  348965 apibinding_controller.go:254] "queueing APIBinding because of APIExport because of APIResourceSchema because of CRD" reconciler="kcp-apibinding" *v1.customresourcedefinition.workspace="system:bound-crds" *v1.customresourcedefinition.namespace="" *v1.customresourcedefinition.name="1252198f-f5c3-4e29-80ca-a640b4916446" *v1.customresourcedefinition.apiVersion="" groupResource="locations.scheduling.kcp.dev" established=false APIResourceSchema="root|v220801-c65c674d4.locations.scheduling.kcp.dev" APIExport="root|scheduling.kcp.dev" key="root|scheduling.kcp.dev"
```

After:

```
I0810 11:41:44.332237  356454 apibinding_controller.go:254] "queueing APIBinding because of APIExport because of APIResourceSchema because of CRD" reconciler="kcp-apibinding" customresourcedefinition.workspace="system:bound-crds" customresourcedefinition.namespace="" customresourcedefinition.name="31e4234e-f93b-44ba-a8f3-360292b73b54" customresourcedefinition.apiVersion="" groupResource="cowboys.wildwest.dev" established=false apiresourceschema.workspace="root:e2e-org-ttkbv:e2e-workspace-wr6sq" apiresourceschema.namespace="" apiresourceschema.name="today.cowboys.wildwest.dev" apiresourceschema.apiVersion="" apiexport.workspace="root:e2e-org-ttkbv:e2e-workspace-wr6sq" apiexport.namespace="" apiexport.name="today-cowboys" apiexport.apiVersion="" key="root:e2e-org-ttkbv:e2e-workspace-p2vg4|cowboys"
```

File: pkg/reconciler/apis/apiexport/apiexport_controller.go [hunk 1]
```diff
@@ -205,19 +204,13 @@ func (c *controller) enqueueAPIExport(obj interface{}) {
...
-	}
-
 	list, err := c.apiExportLister.List(labels.Everything())
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 
-	logger := logging.WithReconciler(klog.Background(), controllerName).WithValues("ClusterWorkspacShard", clusterWorkspaceShardKey)
+	logger := logging.WithObject(logging.WithReconciler(klog.Background(), controllerName), clusterWorkspaceShard.(*tenancyv1alpha1.ClusterWorkspaceShard))
```
Comment (ncdc) [inline pkg/reconciler/apis/apiexport/apiexport_controller.go:213]: (no action required) I'd split this in 2 to make it easier to read

File: pkg/reconciler/apis/apiexport/apiexport_controller.go [hunk 2]
```diff
@@ -243,7 +236,7 @@ func (c *controller) enqueueSecret(obj interface{}) {
 		return
 	}
 
-	logger := logging.WithReconciler(klog.Background(), controllerName).WithValues("Secret", secretKey)
+	logger := logging.WithObject(logging.WithReconciler(klog.Background(), controllerName), obj.(*corev1.Secret))
```
Comment (ncdc) [inline pkg/reconciler/apis/apiexport/apiexport_controller.go:239]: (no action required) I'd split this in 2 to make it easier to read

File: pkg/reconciler/apis/apiexport/apiexport_reconcile.go [hunk 3]
```diff
@@ -110,14 +109,16 @@ func (c *controller) reconcile(ctx context.Context, apiExport *apisv1alpha1.APIE
...
 func (c *controller) ensureSecretNamespaceExists(ctx context.Context, clusterName logicalcluster.Name) {
-	logger := klog.FromContext(ctx).WithValues("Namespace", clusters.ToClusterAwareKey(clusterName, c.secretNamespace))
+	logger := klog.FromContext(ctx)
 	ctx = klog.NewContext(ctx, logger)
 	if _, err := c.getNamespace(clusterName, c.secretNamespace); errors.IsNotFound(err) {
 		ns := &corev1.Namespace{
 			ObjectMeta: metav1.ObjectMeta{
-				Name: c.secretNamespace,
+				Name:        c.secretNamespace,
+				Annotations: map[string]string{logicalcluster.AnnotationKey: clusterName.String()},
```
Comment (ncdc) [inline pkg/reconciler/apis/apiexport/apiexport_reconcile.go:118]: Just making sure: this is only needed for `logging.WithObject` to include the cluster name?

File: pkg/reconciler/apis/apiexport/apiexport_reconcile.go [hunk 4]
```diff
@@ -130,6 +131,7 @@ func (c *controller) createIdentitySecret(ctx context.Context, clusterName logic
 	if err != nil {
 		return err
 	}
+	secret.Annotations[logicalcluster.AnnotationKey] = clusterName.String()
```
Comment (ncdc) [inline pkg/reconciler/apis/apiexport/apiexport_reconcile.go:134]: Just making sure: this is only needed for `logging.WithObject` to include the cluster name?

File: pkg/reconciler/tenancy/clusterworkspacetype/clusterworkspacetype_controller.go [hunk 5]
```diff
@@ -127,27 +127,21 @@ func (c *controller) enqueueClusterWorkspaceType(obj interface{}) {
...
-		return
-	}
-
 	list, err := c.clusterWorkspaceTypeLister.List(labels.Everything())
 	if err != nil {
 		runtime.HandleError(err)
 		return
 	}
 
+	logger := logging.WithObject(logging.WithReconciler(klog.Background(), controllerName), clusterWorkspaceShard.(*tenancyv1alpha1.ClusterWorkspaceShard))
```
Comment (ncdc) [inline pkg/reconciler/tenancy/clusterworkspacetype/clusterworkspacetype_controller.go:136]: (no action required) I'd split this in 2 to make it easier to read

File: pkg/reconciler/apis/apiexport/apiexport_controller.go (hunk 1 above)
Comment (stevekuznetsov) [reply pkg/reconciler/apis/apiexport/apiexport_controller.go:213]: Fairly pervasive at this point - perhaps I can try to make this a builder pattern in a follow-up and fix them all?

File: pkg/reconciler/apis/apiexport/apiexport_reconcile.go (hunk 3 above)
Comment (stevekuznetsov) [reply pkg/reconciler/apis/apiexport/apiexport_reconcile.go:118]: Yes. I'm open to better ideas :|

File: pkg/reconciler/apis/apiexport/apiexport_reconcile.go (hunk 4 above)
Comment (stevekuznetsov) [reply pkg/reconciler/apis/apiexport/apiexport_reconcile.go:134]: Yes. I'm open to better ideas :|

File: pkg/reconciler/apis/apiexport/apiexport_controller.go (hunk 1 above)
Comment (ncdc) [reply pkg/reconciler/apis/apiexport/apiexport_controller.go:213]: SGTM

**Comment by ncdc** [comment]
/approve

@sttts WDYT about https://github.com/kcp-dev/kcp/pull/1730#discussion_r942740842 ?

**Comment by ncdc** [comment]
Happy to merge & iterate too (on the annotation bit)

**Comment by stevekuznetsov** [comment]
@ncdc sure thing - the obvious choice would be `logging.WithClusterObject(obj Object, cluster logicalcluster.Name)` but I'm still not entirely certain why having the logical cluster on the object itself is bad and why we should prefer to pass it out-of-band to clients?

File: pkg/reconciler/kubequota/kubequota_controller.go [hunk 6]
```diff
@@ -167,11 +169,13 @@ func (c *Controller) Start(ctx context.Context, numThreads int) {
...
-	klog.Infof("Starting %s controller", controllerName)
-	defer klog.Infof("Shutting down %s controller", controllerName)
+	logger := logging.WithReconciler(klog.FromContext(ctx), controllerName)
+	ctx = klog.NewContext(ctx, logger)
+	logger.Info("Starting controller")
+	defer logger.Info("Shutting down controller")
 
 	if !cache.WaitForCacheSync(ctx.Done(), c.crdsSynced) {
-		klog.Errorf("Error waiting for CRD informer to sync")
+		logger.Info("error waiting for CRD informer to sync")
```
Comment (ncdc) [inline pkg/reconciler/kubequota/kubequota_controller.go:178]: You sure you want .Info here?

**Pull request merged**